Doc/whatsnew/3.14.rst

```diff
@@ -208,6 +209,29 @@ configuration mechanisms).
 .. seealso::
    :pep:`741`.
 
+.. _whatsnew314-tail-call:
+
+A new tail-calling interpreter
+------------------------------
+
+A new type of interpreter based on tail calls has been added to CPython.
+For certain newer compilers, this interpreter provides
+significantly better performance. Preliminary numbers on our machines suggest
+anywhere from -3% to 30% faster Python code, and a geometric mean of 9-15%
+faster on ``pyperformance`` depending on platform and architecture.
+
+This interpreter currently only works with Clang 19 and newer
+on x86-64 and AArch64 architectures. However, we expect
+that a future release of GCC will support this as well.
+
+This feature is opt-in for now. Based on our own testing,
+this new interpreter only works well when profile-guided optimization is
```

Comment:
Could be good to clarify what "works well" means. Maybe "this new interpreter only improves performance when profile-guided optimization is enabled"?

Reply:
We've only tested and guaranteed it improves perf when PGO is enabled. We can't make guarantees for anything else.